collect_app/src/test/java/org/odk/collect/android/widgets/WidgetFactoryTest.java

```diff
@@ -0,0 +1,345 @@
+import org.odk.collect.android.widgets.items.SelectMultiWidget;
+import org.odk.collect.android.widgets.items.SelectOneImageMapWidget;
+import org.odk.collect.android.widgets.items.SelectOneMinimalWidget;
+import org.odk.collect.android.widgets.items.SelectOneWidget;
+import org.odk.collect.android.widgets.support.FakeQuestionMediaManager;
+import org.odk.collect.android.widgets.support.FakeWaitingForDataRegistry;
+import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+public class WidgetFactoryTest {
+    private WidgetFactory widgetFactory;
+
+    @Before
+    public void setup() {
+        Activity activity = RobolectricHelpers.buildThemedActivity(TestScreenContextActivity.class).get();
+        WaitingForDataRegistry waitingForDataRegistry = new FakeWaitingForDataRegistry();
+        QuestionMediaManager questionMediaManager = new FakeQuestionMediaManager();
+        ActivityAvailability activityAvailability = mock(ActivityAvailability.class);
+
+        widgetFactory = new WidgetFactory(activity,
+                false,
+                false,
+                waitingForDataRegistry,
+                questionMediaManager,
+                null,
+                null,
+                activityAvailability,
+                null);
+    }
+
+    @Test
+    public void testCreatingSelectOneMinimalWidgetWithFastExternalChoices() {
+        QuestionDef questionDef = mock(QuestionDef.class);
+        when(questionDef.getAdditionalAttribute(null, "query")).thenReturn("instance('counties')/root/item[state= /data/state ]");
+
+        FormEntryPrompt prompt = new MockFormEntryPromptBuilder()
+                .withControlType(Constants.CONTROL_INPUT)
+                .withDataType(Constants.DATATYPE_TEXT)
+                .withQuestion(questionDef)
+                .withAppearance("search(‘items’) LisT miNimal something")
+                .build();
+
+        QuestionWidget widget = widgetFactory.createWidgetFromPrompt(prompt);
+        assertThat(widget, instanceOf(SelectOneMinimalWidget.class));
+    }
+
+    @Test
+    public void testCreatingLikertWidgetWithFastExternalChoices() {
+        QuestionDef questionDef = mock(QuestionDef.class);
+        when(questionDef.getAdditionalAttribute(null, "query")).thenReturn("instance('counties')/root/item[state= /data/state ]");
+
+        FormEntryPrompt prompt = new MockFormEntryPromptBuilder()
+                .withControlType(Constants.CONTROL_INPUT)
+                .withDataType(Constants.DATATYPE_TEXT)
+                .withQuestion(questionDef)
+                .withAppearance("search(‘items’) LisT lIkErt something")
```

Comment:
Same comment as above. Why use two external data features? If this is about verifying that appearances after `search()` are honored, no need to also have it be an external select.

Reply:
Same here this test is removed.